src/dune_rules/pkg_rules.ml

```diff
@@ -575,6 +575,7 @@ module Action_expander = struct
       (* values set with withenv *)
       Env.Map.map expander.env ~f:Env_update.string_of_env_values
       |> Env.Map.to_list_map ~f:(fun variable value ->
+        (* TODO why is [package = None]? *)
```

Comment:
> If you use withenv I'd say it doesn't define which package the value belongs to, (withenv ((= FOO bar)) (substitute foo.in foo)) would probably assume the variable is %{FOO}% and not %{package-name:FOO}%.

Might be best to just ask the opam maintainers what's correct. At the end of the day, we just want to replicate what they're doing accurately.

Reply:
This is what the OPAM manpage states:

```
       OPAMVAR_var overrides the contents of the variable var when substituting `%{var}%` strings in `opam` files.

       OPAMVAR_package_var overrides the contents of the variable package:var when substituting `%{package:var}%` strings in `opam` files.
```

These are environment variables, so I've implemented the first form, a variable that is not attached to a package. I could extend the implementation that a variable of the form `PKG_<anything>` will parse as package name. My concern is that this makes sense in `OPAMVAR_package_var` because the prefix makes it clear that we're dealing with packages, but `(withenv ((= LD_LIBRARIES_PRELOAD ...) ...)` would confuse people if `LD` suddenly parsed as package name.